### PR TITLE
emsdk: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/12397
closes https://github.com/conan-io/conan-center-index/issues/12594

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
I've tested it locally on bzip2, and it works well, but it needs few additional modifications to statisfy conan v2 hook. Moreover a true test package with CMakeToolchain is needed.

File: recipes/emsdk/all/test_package/conanfile.py [hunk 1]
```diff
@@ -1,38 +1,46 @@
...
-            self.run("em++ -v", run_environment=True)
+        self.run("emcc -v", env="conanbuild")
+        self.run("em++ -v", env="conanbuild")
 
         # Run the project that was built using emsdk
         if self.settings.os == "Emscripten":
-            test_file = os.path.join("bin", "test_package.js")
-            self.run("node %s" % test_file, run_environment=True)
+            test_file = os.path.join(self.cpp.build.bindirs[0], "test_package.js")
+            self.run(f"node {test_file}", env="conanrun")
```
Comment (SpaceIm) [inline recipes/emsdk/all/test_package/conanfile.py:46]: Should it be `env=conanbuild` since node comes from build context?

**Review by paulocoutinhox — DISMISSED** [review]
OK

**Comment by paulocoutinhox** [comment]
You need put `, encoding='utf-8'` on open function.

**Review by paulocoutinhox — APPROVED** [review]

**Review by paulocoutinhox — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by SpaceIm** [comment]
@prince-chrismc @jgsogo @danimtb could you review please? This PR seems quite critical for @paulocoutinhox

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**